localstack/services/apigateway/invocations.py

```diff
@@ -215,14 +217,28 @@ def apply_response_parameters(invocation_context: ApiInvocationContext):
     int_responses = integration.get("integrationResponses") or {}
     if not int_responses:
         return response
+
+    integration_type_orig = integration.get("type") or integration.get("integrationType") or ""
+    integration_type = integration_type_orig.upper()
+    if integration_type == "AWS_PROXY":
+        LOG.warning("AWS_PROXY integration type should not apply response parameters")
+        return response
+
     entries = list(int_responses.keys())
     return_code = str(response.status_code)
-    if return_code not in entries:
-        if len(entries) > 1:
-            LOG.info("Found multiple integration response status codes: %s", entries)
-            return response
-        return_code = entries[0]
-    response_params = int_responses[return_code].get("responseParameters", {})
+    # Selecting the right integration response
+    # FIXME is this check valid? It should map to the default one or based on the selectionPattern
+    # Not sure if it gets mapped automatically to a method response if it matches the statusCode by chance
+    LOG.debug(
+        "Found multiple integration response status codes: %s, choosing by selection pattern",
+        entries,
+    )
+    selected_integration_response = select_integration_response(return_code, invocation_context)
+    # set status code of integration response
+    response.status_code = selected_integration_response["statusCode"]
+    response_params = int_responses[selected_integration_response["statusCode"]].get(
+        "responseParameters", {}
+    )
```

Comment:
```suggestion
    response_params = selected_integration_response.get("responseParameters") or {}
```
I think we already have the integration response here, we can directly access it. I've changed the check because it might be that the `responseParameters` key is present in the `dict`, but set to `None` or something. With the `or` check, we're sure that we're using a dictionary when calling `.items()` right under.

Reply:
Also a leftover, thanks!